libipl/p10/ipl0.C

```diff
@@ -374,28 +387,137 @@ static int ipl_alignment_check(void)
 	return -1;
 }
 
+static int initialize_and_check_clock_chip(uint8_t& clock_count)
+{
+	struct pdbg_target *clock_target;
+	enum pdbg_target_status status;
+	uint8_t data;
+	int i2c_rc = 0;
+	int rc = 0;
+
+	// initialize output variable.
+	clock_count = 0;
+
+	ipl_log(IPL_DEBUG, "Istep: soft reset clock, and verify clock status register\n");
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+		if(!ipl_is_functional(clock_target)){
+			continue;
+		}
+
+		// clock count has to be incremented even if it failed to initiallize
+		clock_count++;
+
+		status = pdbg_target_probe(clock_target);
+		if(status != PDBG_TARGET_ENABLED){
+			ipl_log(IPL_ERROR, "pdbg_target_probe is failed for clock '%s', with status = %d\n",
```

Comment:
This is not realy probe failed , target is not function, refer previous command related to this error.

Reply:
I didnt get this comment.
can i get rid of pdbg_target_probe(), if i update the functional state and doing ipl_is_functional() in this loop?

Reply:
pdbg_target_probe(clock_target) is required only once for doing hardware access , if you do this at updatehwmodel , if is good for normal ipl. but it is going to fail in individiual istep mode. so better to do just prior to hardware access. 
Also (status != PDBG_TARGET_ENABLED ) doesnt mean probe function failed , if is ay's hw is not ready for use.

Reply:
Done